src/components/markdown-renderer/replace-components/click-shield/click-shield.tsx

```diff
@@ -0,0 +1,122 @@
+import { cypressId } from '../../../../utils/cypress-attribute'
+import { ForkAwesomeIcon } from '../../../common/fork-awesome/fork-awesome-icon'
+
+const log = new Logger('OneClickEmbedding')
+
+interface ClickShieldProps extends PropsWithDataCypressId {
+  onImageFetch?: () => Promise<string>
+  fallbackPreviewImageUrl?: string
+  hoverIcon?: IconName
+  hoverTextI18nKey?: string
+  targetDescription?: string
+  containerClassName?: string
+  fallbackBackgroundColor?: Property.BackgroundColor
+}
+
+/**
+ * Prevents loading of the children elements until the user unlocks the content by e.g. clicking.
+ *
+ * @param containerClassName Additional CSS classes for the complete component
+ * @param onImageFetch A callback that is used to get an URL for the preview image
+ * @param fallbackPreviewImageUrl The URL for an image that should be shown. If onImageFetch is defined then this image will be shown until onImageFetch provides another URL.
+ * @param targetDescription The name of the target service
+ * @param hoverIcon The name of an icon that should be shown in the preview
+ * @param fallbackBackgroundColor A color that should be used if no background image was provided or could be loaded.
+ * @param children The children element that should be shielded.
+ */
+export const ClickShield: React.FC<ClickShieldProps> = ({
+  containerClassName,
+  onImageFetch,
+  fallbackPreviewImageUrl,
+  children,
+  targetDescription,
+  hoverIcon,
+  fallbackBackgroundColor,
+  ...props
+}) => {
+  const [showChildren, setShowChildren] = useState(false)
+  const [previewImageUrl, setPreviewImageUrl] = useState(fallbackPreviewImageUrl)
+  const { t } = useTranslation()
+
+  const doShowChildren = useCallback(() => {
+    setShowChildren(true)
+  }, [])
+
+  useEffect(() => {
+    if (!onImageFetch) {
+      return
+    }
+    onImageFetch()
+      .then((imageLink) => {
+        setPreviewImageUrl(imageLink)
+      })
+      .catch((message) => {
+        log.error(message)
+      })
+  }, [onImageFetch])
+
+  const fallbackBackgroundStyle = useMemo<React.CSSProperties>(
+    () =>
+      !fallbackBackgroundColor
+        ? {}
+        : {
+            backgroundColor: fallbackBackgroundColor
+          },
+    [fallbackBackgroundColor]
+  )
+
+  const previewHoverText = useMemo(() => {
+    return targetDescription ? t('renderer.clickShield.previewHoverText', { target: targetDescription }) : ''
+  }, [t, targetDescription])
+
+  const previewBackground = useMemo(() => {
+    if (previewImageUrl !== undefined) {
+      return (
+        <ProxyImageFrame
+          className={'preview-background embed-responsive-item'}
+          style={fallbackBackgroundStyle}
+          src={previewImageUrl}
+          alt={previewHoverText}
+          title={previewHoverText}
+        />
+      )
+    } else {
+      return <span className={'preview-background embed-responsive-item'} style={fallbackBackgroundStyle} />
+    }
```

Comment:
Simplify by swapping if-else?

```suggestion
    if (!previewImageUrl) {
      return <span className={'preview-background embed-responsive-item'} style={fallbackBackgroundStyle} />
    }
    return (
      <ProxyImageFrame
        className={'preview-background embed-responsive-item'}
        style={fallbackBackgroundStyle}
        src={previewImageUrl}
        alt={previewHoverText}
        title={previewHoverText}
      />
    )
```